glib/src/main_context_futures.rs

```diff
@@ -18,15 +24,15 @@ use crate::SourceId;
 // Wrapper around Send Futures and non-Send Futures that will panic
 // if the non-Send Future is polled/dropped from a different thread
 // than where this was created.
-enum FutureWrapper {
-    Send(FutureObj<'static, ()>),
-    NonSend(ThreadGuard<LocalFutureObj<'static, ()>>),
+enum FutureWrapper<T> {
+    Send(FutureObj<'static, T>),
+    NonSend(ThreadGuard<LocalFutureObj<'static, T>>),
 }
```

Comment:
This (for completeness) should probably get a `Send` impl now

Reply:
I don't think it's needed, this is not a public type

Reply:
Not needed but it would document the API a bit more